### PR TITLE
Bug 2106366, Bug 2106377: Use displayname for PHCR in the catalog page

### PR DESCRIPTION
**Pull request by debsmita1**
**Fixes:**
https://issues.redhat.com/browse/OCPBUGSM-46711
https://bugzilla.redhat.com/show_bug.cgi?id=2106366

**Root analysis:**
The display name for PHCR is not used as the project scoped chart repositories are not fetched

**Solution description:**
- Added input field for display name
- Added ProjectHelmChartRepository type in the resource watcher 

**GIF:**
https://user-images.githubusercontent.com/22490998/179819227-2bda705d-6f9b-4b97-8c21-131e1a0b3177.mov

**Test coverage:**
Added e2e test
Screenshot 2022-07-19 at 11 32 23 PM

### CODE REVIEW

**Comment by sanketpathak** [comment]
Tests are running green
Screenshot from 2022-07-20 12-12-43

File: frontend/packages/dev-console/integration-tests/features/addFlow/helm-chart-repositories.feature [hunk 1]
```diff
@@ -38,4 +38,14 @@ Feature: Install the Helm Release
               And user enters Description as "test"
               And user enters URL as "https://raw.githubusercontent.com/IBM/charts/master/repo/community/index.yaml"
               And user clicks on Create button
-             Then user can see "helm-test1" under Chart Repositories in Helm Charts catalog page
+               Then user can see "helm-test1" for resource "helm-test1" and type "projecthelmchartrepository" under Chart Repositories in Helm Charts catalog page
+
+         @regression
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/addFlow/helm-chart-repositories.feature:43]: Can you add epic tag as well
```suggestion
         @regression @odc-6685
```

Comment (debsmita1) [reply frontend/packages/dev-console/integration-tests/features/addFlow/helm-chart-repositories.feature:43]: @sanketpathak but the form was introduced in ODC-6452

**Comment by jerolimov** [comment]
/retitle Bug 2106366, Bug 2106377: Use displayname for PHCR in the catalog page
/retest

**Review by jerolimov — APPROVED** [review]
Tested this and it works fine. Tested also the case when I removed the name from the YAML manually.

/approve

When this feedback is resolved I would like to add lgtm as well.

File: frontend/packages/dev-console/integration-tests/features/addFlow/helm-chart-repositories.feature (hunk 1 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/addFlow/helm-chart-repositories.feature]: It's related to the test scenario created with the epic

**Pull request merged**